projects/packages/analyzer/src/diff-generator.php

```diff
@@ -40,16 +40,33 @@ static function get_differences( $new_path, $old_path ) {
 	}
 
 	static function get_warnings( $folder_name, $differences, $excludes ) {
+		$warnings_folder = dirname( __DIR__ ) . '/output/warnings/';
+		$invocations_folder = dirname( __DIR__ ) . '/output/invocations/';
+
 		echo "Looking for invocations in:\n${folder_name}\n\n";
 		$invocations = new Invocations();
 		$invocations->scan( $folder_name, $excludes );
-		$invocations->save( dirname( __DIR__ ) . '/output/invocations/' . basename( $folder_name ) . '.json', false );
+		$invocations->save( $invocations_folder . basename( $folder_name ) . '.json', false );
 
 		echo "Generate warnings\n";
 		$warnings = new Warnings();
 		$warnings->generate( $invocations, $differences );
 		$warnings->output();
-		$warnings->save_json( dirname( __DIR__ ) . '/output/warnings/' . basename( $folder_name ) . '.json', false );
+		$warnings->save_json( $warnings_folder . basename( $folder_name ) . '.json', false );
+	}
+
+	static function cleanup() {
+		$diff_folder = dirname( __DIR__ ) . '/output/differences/';
+		$warnings_folder = dirname( __DIR__ ) . '/output/warnings/';
+		$invocations_folder = dirname( __DIR__ ) . '/output/invocations/';
+		echo "Removing " . $diff_folder . "\n";
+		system( "rm -rf " . escapeshellarg( $diff_folder ) );
```

Comment:
Even though it's more code, I'd prefer something like https://github.com/Automattic/jetpack/blob/35cfa36693e61f36ccaf3aba4b0a78ef34c8b887/projects/packages/changelogger/tests/php/includes/src/TestCase.php#L104-L115 over shelling out to `rm -rf`. It seems less likely to screw up if something winds up being named oddly.

Reply:
Yeah, the shell approach is ugly. thanks for the snippet!